js/mixins/dom_iterable_test.ts

```diff
@@ -0,0 +1,78 @@
+// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+import { test, assert, assertEqual } from "../test_util.ts";
+import { DomIterableMixin } from "./dom_iterable.ts";
+
+function setup() {
+  const dataSymbol = Symbol("data symbol");
+  class Base {
+    private [dataSymbol] = new Map<string, number>();
+
+    constructor(
+      data: Array<[string, number]> | IterableIterator<[string, number]>
+    ) {
+      for (const [key, value] of data) {
+        this[dataSymbol].set(key, value);
+      }
+    }
+  }
+
+  return {
+    Base,
+    DomIterable: DomIterableMixin<string, number, typeof Base>(Base, dataSymbol)
+  };
+}
+
+test(function testDomIterable() {
+  // tslint:disable-next-line:variable-name
+  const { DomIterable, Base } = setup();
+
+  const fixture: Array<[string, number]> = [["foo", 1], ["bar", 2]];
+
+  const domIterable = new DomIterable(fixture);
+
+  assertEqual(Array.from(domIterable.entries()), fixture);
+  assertEqual(Array.from(domIterable.values()), [1, 2]);
+  assertEqual(Array.from(domIterable.keys()), ["foo", "bar"]);
+
+  let result: Array<[string, number]> = [];
+  for (const [key, value] of domIterable) {
+    assert(key != null);
+    assert(value != null);
+    result.push([key, value]);
+  }
+  assertEqual(fixture, result);
+
+  result = [];
+  const scope = {};
+  function callback(value, key, parent) {
+    assertEqual(parent, domIterable);
+    assert(key != null);
+    assert(value != null);
+    assert(this === scope);
```

Comment:
This should use assertEqual
```suggestion
    assertEqual(this, scope);
```

Reply:
`assertEqual()` does loose comparison on objects.  Here we want strict equality to ensure that the passed scope is the actual scope, not something that just looks like the passed scope.

https://github.com/denoland/deno/blob/c4bddc4651f583241193b80634a4f22abb02d582/js/testing/util.ts#L41-L64

With the additional scope tests, it a bit of a gamble that the references might not be actual references, but because non-objects are coerced into objects, it is impossible to get a reference to the actual scope.